duplex_http_call.go

```diff
@@ -42,13 +42,11 @@ type duplexHTTPCall struct {
 	requestBodyReader *io.PipeReader
 	requestBodyWriter *io.PipeWriter
 
-	sendRequestOnce sync.Once
-	responseReady   chan struct{}
-	request         *http.Request
-	response        *http.Response
-
-	errMu sync.Mutex
-	err   error
+	requestSent   bool
+	responseReady sync.WaitGroup
```

Comment:
`responseReady` as a `sync.WaitGroup`.

Reply:
They are equivalent but `sync.WaitGroup` I thought was more idiomatic and avoids needing to allocate a channel. Can revert.